### DISCUSSION TITLE
How to do  text compression

### DISCUSSION THREAD

**Discussion by s1n7ax** [leptos-rs/leptos#2591] [Q&A] [answered]
I have reached almost 100% in performance but there are few things lacking.

I'm facing two issues:
- Text content is not served compressed (As shown in the image 1)
- Javascript is not minified (As shown in the image 2,3,4)

I'm using release mode when deploying:
https://github.com/s1n7ax/my-website-ssr/blob/8a081398bfef4d82ac468153dc6306c78d2a4512/Dockerfile?plain=1#L19-L21

Is there anything I could do to enable text compression & js minifying working?

![image](https://github.com/leptos-rs/leptos/assets/18459807/c2c02d9b-4861-4843-a3d2-fb4d8f9146f3)
![image](https://github.com/leptos-rs/leptos/assets/18459807/a1ef4d16-48a2-4f11-8867-b7609d0edb8b)
![image](https://github.com/leptos-rs/leptos/assets/18459807/bbdf764b-abca-4f0d-98b0-91bba9af0915)
![image](https://github.com/leptos-rs/leptos/assets/18459807/f7059a14-f608-4638-9dfe-c4060d3d24f9)

**Comment by gbj** [comment, marked as answer]
Sure. You have the ability to do both of these and they're both totally independent of Leptos. Just add a `tower_http` [`CompressionLayer`](https://docs.rs/tower-http/0.5.2/tower_http/compression/index.html) in your `main.rs`, and use the JS minifier of your choice during the build process.

**Reply by s1n7ax** [reply]
@gbj Got it. Thanks. Leptos isn't easy (for someone coming from JavaScript world) but I'm loving it. Amazing work. I watch your youtube also.
